crates/metastore/src/database.rs

```diff
@@ -1302,10 +1302,10 @@ impl BuiltinCatalog {
             let schema_id = schema_names
                 .get(DEFAULT_SCHEMA)
                 .ok_or_else(|| MetastoreError::MissingNamedSchema(DEFAULT_SCHEMA.to_string()))?;
-            let mut entry = func.as_function_entry(oid, *schema_id);
-            entry.runtime_preference = func.runtime_preference();
 
```

Comment:
previously the only reason we needed different for loops for `table_funcs()` and `scalar_funcs()` was for the handling of `runtime_preference`. Since that is removed, we should follow up with cleaning up that logic.

Reply:
Will open another PR for this (since it would require some more review).